src/targets/gpu/target.cpp

```diff
@@ -179,6 +179,7 @@ std::vector<pass> target::get_passes(migraphx::context& gctx, const compile_opti
         dead_code_elimination{},
         enable_pass(not gpu::gfx_has_fp8ocp_intrinsics() and gpu::gfx_has_fp8fnuz_intrinsics(), fp8_ocp_to_fnuz{}),
         enable_pass(not gpu::gfx_has_fp8ocp_intrinsics() and gpu::gfx_has_fp8fnuz_intrinsics(), dead_code_elimination{}),
+        enable_pass(not gpu::gfx_has_bf16_intrinsics(), eliminate_data_type{{migraphx::shape::bf16_type}, shape::type_t::float_type}),
```

Comment:
We shouldn't run another pass, this should just conditionally remove bf16 from the unsupported types.